doc/source/reference/array_api.rst

```diff
@@ -0,0 +1,787 @@
+.. _array_api:
+
+*********
+Array API
+*********
+
```

Comment:
Let's add a `.. note::` here about the implementation in NumPy currently being experimental.